### ISSUE TITLE
r504-x64 build 21975 bug

### ISSUE THREAD

**Issue by Havokdan** [pavel-pimenov/flylinkdc-r5xx#1728]
In this latest update, I can not stay active, look I'm active for over six years with the same configuration.

Windows 10 x64

**Comment by pavel-pimenov** [comment]
![изображение](https://user-images.githubusercontent.com/920705/67157233-0dfb5400-f332-11e9-804a-3c2d64598bbe.png)

**Comment by Havokdan** [comment]
Since the last update, the TCP icon is not green, turns gray. And yes, it is set up properly, so I have another client DC ++ here (AirDC ++) that is functioning normally and the very Flylinkdc ++ always worked correctly before.

**Comment by pavel-pimenov** [comment]
Push "test port..." + screenshot

**Comment by Havokdan** [comment]
![image](https://user-images.githubusercontent.com/6549729/67163160-18bde500-f342-11e9-9faf-c01abdc5c0ce.png)

**Comment by Havokdan** [comment]
Any news? Can I go back to the previous version?

**Comment by Havokdan** [comment]
> All build
> https://cloud.mail.ru/public/9J3w/6A3wMMV2R
> 
> Extract and replace FlylinkDC*.exe

Thank you, it worked, I'm off the updates and will also keep a backup of this latest version that works.

**Comment by pavel-pimenov** [comment]
build-xxxx ? + screenshot ?

**Comment by Havokdan** [comment]
r504-x64 build 21969:

![image](https://user-images.githubusercontent.com/6549729/67418818-e3afcd80-f5a1-11e9-9bdf-7366cf3eb6d3.png)

**Comment by pavel-pimenov** [comment]
upnp -> manual?
![изображение](https://user-images.githubusercontent.com/920705/67426923-bab02c80-f5e3-11e9-8cba-015078c14ba5.png)

**Comment by Havokdan** [comment]
Upnp (However, both have always worked for me for years without problems, not to mention that the other customer use in parallel, AirDC ++ continues working normally):

![image](https://user-images.githubusercontent.com/6549729/67432217-431ad700-f5bc-11e9-81db-bb10504260e6.png)

**Comment by Havokdan** [comment]
For the record, I tested all new versions since then and the problem continues.

**Comment by pavel-pimenov** [comment]
build-22084
![изображение](https://user-images.githubusercontent.com/920705/68890706-8c2ff800-0730-11ea-8f44-2b8ecc210f14.png)

**Comment by Havokdan** [comment]
Same...

![image](https://user-images.githubusercontent.com/6549729/68891678-1875f700-0700-11ea-8ca9-f5f227915efc.png)
last version that worked was 21969.

**Comment by pavel-pimenov** [comment]
hm upnp  - error

1.  Model router?
2.  show Logs\System.log ? (mask = *portmap_alert*)

![изображение](https://user-images.githubusercontent.com/920705/68892375-e7afb500-0733-11ea-85f3-cc9b787f3fec.png)

**Comment by Havokdan** [comment]
Router = D-Link  DSL-2740E

Log (Partial):

`57:00] [Step]  Error POST query [4000 ms]
[2019-11-14 16:57:08] Você tem o poder de um operador neste hub dchub://quadrinhos.duckdns.org:6000
[2019-11-14 16:57:21] portmap_error_alert: could not map port using UPnP: no router found info:portmap_error_alert index = 0
[2019-11-14 16:57:21] portmap_error_alert: could not map port using UPnP: no router found info:portmap_error_alert index = 1
[2019-11-14 16:57:21] portmap_error_alert: could not map port using UPnP: no router found info:portmap_error_alert index = 2
[2019-11-14 16:57:21] portmap_error_alert: could not map port using UPnP: no router found info:portmap_error_alert index = 3
[2019-11-14 16:57:21] portmap_error_alert: could not map port using UPnP: no router found info:portmap_error_alert index = 4
[2019-11-14 16:57:21] [Step]  HttpSendRequest error O tempo limite da operação foi atingido[error: 12002] [4001 ms]
[2019-11-14 16:57:21] [Step]  Use next mirror server: media.fly-server.ru:37015 [4002 ms]
[2019-11-14 16:57:25] [Step]  HttpSendRequest error O tempo limite da operação foi atingido[error: 12002] [4001 ms]
[2019-11-14 16:57:25] [Step]  Error POST query [4001 ms]
[2019-11-14 16:57:27] [Step] [GetIP] Download : http://checkip.dyndns.com IP = 200.181.35.26 [1836 ms]
[2019-11-14 16:57:27] [Stop] [GetIP] [0 ms, Total: 1836 ms]
[2019-11-14 16:58:15] MoveFile OK C:\Users\Daniel\AppData\Roaming\FlylinkDC++\DCPlusPlus.xml.tmp -> C:\Users\Daniel\AppData\Roaming\FlylinkDC++\DCPlusPlus.xml
[2019-11-14 16:58:15] [Stop] [Rename] [1 ms, Total: 1 ms]
[2019-11-14 16:58:25] Você tem o poder de um operador neste hub dchub://quadrinhos.duckdns.org:6000
[2019-11-14 16:58:35] [Step]  HttpSendRequest error O tempo limite da operação foi atingido[error: 12002] [4002 ms]
[2019-11-14 16:58:35] [Step]  Use next mirror server: media.fly-server.ru:37015 [4002 ms]
[2019-11-14 16:58:39] [Step]  HttpSendRequest error O tempo limite da operação foi atingido[error: 12002] [4000 ms]
[2019-11-14 16:58:39] [Step]  Error POST query [4000 ms]
[2019-11-14 16:58:42] [Step] [GetIP] Download : http://checkip.dyndns.com IP = 200.181.35.26 [2530 ms]
[2019-11-14 16:58:42] [Stop] [GetIP] [0 ms, Total: 2530 ms]
[2019-11-14 16:59:30] MoveFile OK C:\Users\Daniel\AppData\Roaming\FlylinkDC++\DCPlusPlus.xml.tmp -> C:\Users\Daniel\AppData\Roaming\FlylinkDC++\DCPlusPlus.xml
[2019-11-14 16:59:30] [Stop] [Rename] [1 ms, Total: 1 ms]
[2019-11-14 16:59:33] [Stop] [Core shutdown] [108 ms, Total: 108 ms]
[2019-11-14 16:59:33] MoveFile OK C:\Users\Daniel\AppData\Roaming\FlylinkDC++\Favorites.xml.tmp -> C:\Users\Daniel\AppData\Roaming\FlylinkDC++\Favorites.xml
[2019-11-14 16:59:33] [Stop] [Rename] [1 ms, Total: 1 ms]
[2019-11-14 16:59:33] Stop alert handler...
[2019-11-14 16:59:33] Alert handler stopped
[2019-11-14 16:59:33] MoveFile OK C:\Users\Daniel\AppData\Roaming\FlylinkDC++\DCPlusPlus.xml.tmp -> C:\Users\Daniel\AppData\Roaming\FlylinkDC++\DCPlusPlus.xml
[2019-11-14 16:59:33] [Stop] [Rename] [0 ms, Total: 0 ms]
[2019-11-14 17:01:25] Add torrent DHT router: router.bittorrent.com:6881
[2019-11-14 17:01:25] Add torrent DHT router: router.utorrent.com:6881
[2019-11-14 17:01:25] Add torrent DHT router: dht.transmissionbt.com:6881
[2019-11-14 17:01:25] Add torrent DHT router: router.bitcomet.com:6881
[2019-11-14 17:04:01] portmap_error_alert: could not map port using UPnP: no router found info:portmap_error_alert index = 0
[2019-11-14 17:04:01] portmap_error_alert: could not map port using UPnP: no router found info:portmap_error_alert index = 1
[2019-11-14 17:04:01] portmap_error_alert: could not map port using UPnP: no router found info:portmap_error_alert index = 2
[2019-11-14 17:04:01] portmap_error_alert: could not map port using UPnP: no router found info:portmap_error_alert index = 3
[2019-11-14 17:04:01] portmap_error_alert: could not map port using UPnP: no router found info:portmap_error_alert index = 4
`

**Comment by pavel-pimenov** [comment]
DSL-2740E enable upnp ?

![изображение](https://user-images.githubusercontent.com/920705/68893894-298e2a80-0737-11ea-90e0-92b4378ed4ec.png)

**Comment by Havokdan** [comment]
No, he had even forgotten that there was disabled for months, I enabled and restarted the windows, but the problem persists.

**Comment by pavel-pimenov** [comment]
DSL-2740E - show system log?

3.4.5 System Log
Choose Maintenance > System Log. The page shown in the following figure
appears. This section when enabled allows the system to begin logging events
based on the selected log level...

**Comment by Havokdan** [comment]
[logfile.log](https://github.com/pavel-pimenov/flylinkdc-r5xx/files/3848919/logfile.log)

**Comment by Havokdan** [comment]
![Sem título](https://user-images.githubusercontent.com/6549729/70706178-33fb0000-1cb4-11ea-9e03-5065836ad019.png)
Does this have any relevance?

**Comment by pavel-pimenov** [comment]
Error POST - fixed

**Comment by Havokdan** [comment]
I switched from broadband here, from adsl to fiber optics and with many other adversities, being able to stay active, if you want to close the problem, I'm ok. Emphasizing that while using adsl, you are unable to become more active after the aforementioned version.

**Comment by Havokdan** [comment]
Yesterday, out of nowhere the problem occurred again in the last version, I did not make any changes to my applications, as well as the hardware, but flylinkdc is not active at all, I tested the same ports on Airdc ++ and it worked normally, I will stay there for while and see if there occurs too.

**Comment by Havokdan** [comment]
![Sem título](https://user-images.githubusercontent.com/6549729/82688366-26687f80-9c2f-11ea-8647-cb6ee3260f44.png)
